foursight_core/react/api/envs.py

```diff
@@ -0,0 +1,82 @@
+import copy
+from typing import Optional
+import os
+from dcicutils import ff_utils
+
+
+class Envs:
+
+    def __init__(self, known_envs: list):
+        # This known_envs should be the list of annotated environment name objects
+        # as returned by app_utils.get_unique_annotated_environment_names, where each
+        # object contains these fields: name, short_name, full_name, public_name, foursight_name
+        self.known_envs = known_envs
+
+    def get_known_envs(self) -> list:
+        return self.known_envs
+
+    def get_known_envs_with_gac_names(self, gac) -> list:
+        known_envs = copy.deepcopy(self.get_known_envs())
+        for known_env in known_envs:
+            known_env["gac_name"] = gac.get_gac_name(known_env["full_name"])
+        return known_envs
+
+    def get_default_env(self) -> str:
+        return os.environ.get("ENV_NAME", "no-default-env")
+
+    def is_known_env(self, env: str) -> bool:
+        return self.find_known_env(env) is not None
+
+    def find_known_env(self, env: str) -> Optional[dict]:
+        if not env:
+            return None
+        env = env.lower()
```

Comment:
This .lower() is suspect.

@willronchetti do you have an opinion here?

Reply:
I typically use all lowercase for env names... In most cases this will probably do nothing

Reply:
My point is they are case-sensitive. You can't use uppercase environment names elsewhere, so no point in pretending.